pkg/ddevapp/router_test.go

```diff
@@ -266,3 +269,140 @@ func TestDisableHTTP2(t *testing.T) {
+	site1 := testcommon.TestSite{
+		Name:                          "TestEphemeralPort1", // Drupal D7
+		SourceURL:                     "https://ftp.drupal.org/files/projects/drupal-7.90.tar.gz",
+		ArchiveInternalExtractionPath: "drupal-7.90/",
+		FilesTarballURL:               "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d7test-7.59.files.tar.gz",
+		DBTarURL:                      "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d7test-7.87-db.tar.gz",
+		FullSiteTarballURL:            "",
+		Dir:                           testcommon.CreateTmpDir(t.Name() + "_1"),
+		Docroot:                       "",
+		Type:                          nodeps.AppTypeDrupal7,
+		Safe200URIWithExpectation:     testcommon.URIWithExpect{URI: "/README.txt", Expect: "Drupal is an open source content management platform"},
+		DynamicURI:                    testcommon.URIWithExpect{URI: "/node/1", Expect: "D7 test project, kittens edition"},
+		FilesImageURI:                 "/sites/default/files/field/image/kittens-large.jpg",
+		FullSiteArchiveExtPath:        "docroot/sites/default/files",
+	}
+
+	site2 := testcommon.TestSite{
+		Name:                          "TestEphemeralPort2",
+		SourceURL:                     "https://ftp.drupal.org/files/projects/drupal-8.9.20.tar.gz",
+		ArchiveInternalExtractionPath: "drupal-8.9.20/",
+		FilesTarballURL:               "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d8_umami.files.tar.gz",
+		FilesZipballURL:               "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d8_umami.files.zip",
+		DBTarURL:                      "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d8_umami.sql.tar.gz",
+		DBZipURL:                      "https://github.com/ddev/ddev_test_tarballs/releases/download/v1.1/d8_umami.sql.zip",
+		FullSiteTarballURL:            "",
+		Dir:                           testcommon.CreateTmpDir(t.Name() + "_2"),
+		Type:                          nodeps.AppTypeDrupal,
+		Docroot:                       "",
+		Safe200URIWithExpectation:     testcommon.URIWithExpect{URI: "/README.txt", Expect: "Drupal is an open source content management platform"},
+		DynamicURI:                    testcommon.URIWithExpect{URI: "/node/2", Expect: "Vegan chocolate and nut brownies"},
+		FilesImageURI:                 "/sites/default/files/vegan-chocolate-nut-brownies.jpg",
+	}
+
+	app, err := ddevapp.NewApp(site1.Dir, false)
+	require.NoError(t, err)
+	app2, err := ddevapp.NewApp(site2.Dir, false)
+	require.NoError(t, err)
+
+	// Configure both apps to use the same target ports. Keep original configured ports for undoing the configuration later.
+	appHTTPPort, appHTTPSPort := app.RouterHTTPPort, app.RouterHTTPSPort
+	app.RouterHTTPPort, app.RouterHTTPSPort = targetHTTPPort, targetHTTPSPort
+	app2HTTPPort, app2HTTPSPort := app.RouterHTTPPort, app.RouterHTTPSPort
+	app2.RouterHTTPPort, app2.RouterHTTPSPort = targetHTTPPort, targetHTTPSPort
```

Comment:
"short variable declaration", I had never seen this before, initializing 2 things at once. I'm not sure it's useful here though :)  I think an array or map assignment might be simpler. Not sure.